enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1864,16 +1864,34 @@ func (p *PebbleCache) newCDCCommitedWriteCloser(ctx context.Context, fileRecord
 		shouldCompress: shouldCompress,
 		isCompressed:   isCompressed,
 	}
+	var wc, decompressor io.WriteCloser
+	wc = cdcw
+
+	if isCompressed {
+		// If the bytes being written are compressed, we decompress them in
+		// order to generate CDC chunks, then compress those chunks.
+		decompressor, err = compression.NewZstdDecompressor(cdcw)
+		if err != nil {
+			return nil, err
+		}
+		wc = decompressor
+	}
 
 	chunker, err := chunker.New(ctx, p.averageChunkSizeBytes, cdcw.writeChunk)
 	if err != nil {
 		return nil, err
 	}
 	cdcw.chunker = chunker
 
-	cwc := ioutil.NewCustomCommitWriteCloser(cdcw)
+	cwc := ioutil.NewCustomCommitWriteCloser(wc)
 	cwc.CloseFn = db.Close
 	cwc.CommitFn = func(bytesWritten int64) error {
+		if decompressor != nil {
```

Comment:
Is it necessary to close the decompressor here? It will be closed when Close() is called on  the CustomCommitWriteCloser.

Reply:
Yes. If the decompressor is not closed; then the decompressor can write to cdcw when cdcw is already closed.